### PR TITLE
Patch to enhance Bonding test

### PR DESCRIPTION
**Pull request by hathyaga**
The enhancement contains an option to create a Bonding Peer
interface if needed and failure check added when all interfaces
are down

Signed-off-by: Harsha Thyagaraja <harshkid@linux.vnet.ibm.com>

### CODE REVIEW

**Comment by hathyaga** [comment]
[Bonding_joblog.txt](https://github.com/avocado-framework-tests/avocado-misc-tests/files/1014024/Bonding_joblog.txt)

File: io/net/bonding.py [hunk 1]
```diff
@@ -110,13 +111,17 @@ def setUp(self):
             self.net_mask.append(mask)
         self.bonding_slave_file = "/sys/class/net/%s/bonding/slaves"\
                                   % self.bond_name
+        self.peer_bond_needed = self.params.get("peer_bond_needed",
+                                                default=False)
+        self.peer_wait_time = self.params.get("peer_wait_time", default=5)
+        self.sleep_time = self.params.get("sleep_time", default=5)
```
Comment (narasimhan-v) [inline io/net/bonding.py]: self.sleep_time = int(self.params.get("sleep_time", default=5))

File: io/net/bonding.py [hunk 2]
```diff
@@ -221,9 +253,13 @@ def bond_setup(self, arg1, arg2):
                     self.fail("unable to interface up")
             cmd = "ifconfig %s %s netmask %s up"\
                   % (self.bond_name, self.host_ips[0], self.net_mask[0])
-            if process.system(cmd, shell=True, ignore_status=True) != 0:
-                self.fail("bond setup command failed in local machine")
-            time.sleep(5)
+            for i in range(0, 600, 60):
+                if process.system(cmd, shell=True, ignore_status=True) != 0:
```
Comment (narasimhan-v) [inline io/net/bonding.py]: Not needed to run the command every minute. This should have been before the `for` loop

**Comment by hathyaga** [comment]
Closing this PR as a new PR has been sent: https://github.com/avocado-framework-tests/avocado-misc-tests/pull/513

**Pull request closed without merging**